app/common/data/models/__init__.py

```diff
@@ -74,4 +76,49 @@ class CollectionSchema(BaseModel):
     created_by_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("user.id"))
     created_by: Mapped[User] = relationship("User")
 
+    sections: Mapped[list["Section"]] = relationship(
+        "Section", lazy=True, order_by="Section.order", collection_class=ordering_list("order", count_from=1)
+    )
+
+    @property
+    def slug(self) -> str:
+        return slugify(self.name)
+
     __table_args__ = (UniqueConstraint("name", "grant_id", "version", name="uq_schema_name_version_grant_id"),)
+
+
+class Section(BaseModel):
+    __tablename__ = "section"
+
+    title: Mapped[str]
+    order: Mapped[int]
+
+    collection_schema_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("collection_schema.id"))
+    collection_schema: Mapped[CollectionSchema] = relationship("CollectionSchema", back_populates="sections")
+
+    forms: Mapped[list["Form"]] = relationship(
+        "Form", lazy=True, order_by="Form.order", collection_class=ordering_list("order", count_from=1)
+    )
+
+    __table_args__ = (
+        UniqueConstraint("order", "collection_schema_id", name="uq_section_order_collection_schema", deferrable=True),
+        UniqueConstraint("title", "collection_schema_id", name="uq_section_title_collection_schema"),
+    )
+
+
+class Form(BaseModel):
+    __tablename__ = "form"
+
+    title: Mapped[str]
+    order: Mapped[int]
+    slug: Mapped[str]
```

Comment:
Also interesting that we have a DB-level slug here, but not for collections - any specific reason for the different implementations?